db/external_sst_file_basic_test.cc

```diff
@@ -1816,51 +1816,164 @@ TEST_F(ExternalSSTFileBasicTest, IngestWithTemperature) {
-  options.create_if_missing = true;
-  options.compaction_style = CompactionStyle::kCompactionStyleUniversal;
-  DestroyAndReopen(options);
-  {
-    const Snapshot* snapshot = db_->GetSnapshot();
-    ManagedSnapshot snapshot_guard(db_, snapshot);
-    IngestExternalFileOptions ifo;
-    ifo.fail_if_not_bottommost_level = true;
-    ifo.snapshot_consistency = true;
-    const Status s = db_->IngestExternalFile({file_path}, ifo);
-    ASSERT_TRUE(s.IsTryAgain());
-  }
+  std::string file_path2 = sst_files_dir_ + std::to_string(2);
 
-  // Test level compaction
-  options.compaction_style = CompactionStyle::kCompactionStyleLevel;
-  options.num_levels = 2;
-  DestroyAndReopen(options);
-  ASSERT_OK(db_->Put(WriteOptions(), "a", "dontcare"));
-  ASSERT_OK(db_->Put(WriteOptions(), "c", "dontcare"));
-  ASSERT_OK(db_->Flush(FlushOptions()));
+  ASSERT_OK(sfw.Open(file_path2));
+  ASSERT_OK(sfw.Put("c05", "dontcare"));
+  ASSERT_OK(sfw.Put("c15", "dontcare"));
+  ASSERT_OK(sfw.Finish());
 
-  ASSERT_OK(db_->Put(WriteOptions(), "b", "dontcare"));
-  ASSERT_OK(db_->Put(WriteOptions(), "d", "dontcare"));
-  ASSERT_OK(db_->Flush(FlushOptions()));
+  // Vary compaction style
+  for (CompactionStyle compaction_style :
+       {CompactionStyle::kCompactionStyleUniversal,
+        CompactionStyle::kCompactionStyleLevel}) {
+    SCOPED_TRACE("compaction_style=" + std::to_string(compaction_style));
+    // Test with a basic DB and snapshot_consistency
+    options.create_if_missing = true;
+    options.compaction_style = compaction_style;
+    options.num_levels = 7;
+    for (bool snapshot_consistency : {false, true}) {
+      SCOPED_TRACE("snapshot_consistency=" +
+                   std::to_string(snapshot_consistency));
+      DestroyAndReopen(options);
+      ASSERT_OK(db_->Put(WriteOptions(), "a00", "dontcare"));
+      ASSERT_OK(db_->Flush(FlushOptions()));
+
+      const Snapshot* snapshot = db_->GetSnapshot();
+      ManagedSnapshot snapshot_guard(db_, snapshot);
+      IngestExternalFileOptions ifo;
+      ifo.fail_if_not_bottommost_level = true;
+      ifo.snapshot_consistency = snapshot_consistency;
+      const Status s = db_->IngestExternalFile({file_path}, ifo);
+      if (snapshot_consistency) {
+        if (compaction_style == CompactionStyle::kCompactionStyleUniversal) {
+          ASSERT_TRUE(s.IsTryAgain());
+        } else {
+          // FIXME: seems to be broken, or am I misunderstanding the
+          // snapshot_consistency option?
+          ASSERT_OK(s);  // WRONG
+        }
+      } else {
+        if (compaction_style == CompactionStyle::kCompactionStyleLevel) {
+          ASSERT_OK(s);
+        } else {
+          // FIXME: for unknown reason, with universal compaction, ingesting
+          // into an empty level is not allowed. (In production code,
+          // compaction_style == kCompactionStyleUniversal -> continue.)
+          ASSERT_TRUE(s.IsTryAgain());  // WRONG
```

Comment:
It doesn't try because base_level is -1 in universal compaction: https://github.com/facebook/rocksdb/blob/226ee25d300c25c6ba4ec4050a69b8d48a6d21f3/db/external_sst_file_ingestion_job.cc#L864-L865

Seems like a bug